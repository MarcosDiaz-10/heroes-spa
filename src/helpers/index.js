export * from './getWordsFromString';
export * from './getQueryParams';