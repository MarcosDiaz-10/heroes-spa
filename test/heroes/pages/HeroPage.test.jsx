import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter, Route, Routes, useNavigate } from "react-router-dom"
import { HeroPage } from "../../../src/heroes"
const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => {
    return{
        ...jest.requireActual('react-router-dom'),
        useNavigate: () => mockUseNavigate
    }
})

describe('Pruebas HeroPage', () => { 

    test('debe mostrar el componente por defecto cuando existe el hero', () => { 

        const { container } = render(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path="hero/:id" element={ <HeroPage/>}/>
                    <Route path="marvel" element={ <h1>Marvel</h1> }/>
                </Routes>             
            </MemoryRouter>
        )
        
        expect( container ).toMatchSnapshot();
        

     })

    test('debe mostrar la imagen cuando el hero existe', () => { 

        render(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path="hero/:id" element={ <HeroPage/>}/>
                    <Route path="marvel" element={ <h1>Marvel</h1> }/>
                </Routes>             
            </MemoryRouter>
        )
        
        const img = screen.getByRole('img');
        expect( img.src ).toContain('marvel-spider')
        
     })

    test('debe navegar a marvel cuando no hay hero', () => { 

        render(
            <MemoryRouter initialEntries={['/hero/marvel-spider2']}>
                <Routes>
                    <Route path="hero/:id" element={ <HeroPage/>}/>
                    <Route path="marvel" element={ <h1>Marvel</h1> }/>
                </Routes>             
            </MemoryRouter>
        )
        
        const marvelPage = screen.getByText('Marvel');
        expect( marvelPage ).toBeTruthy();
        

     })

    test('debe llamar la funcion de navegar al presionar el boton de regresar', () => { 

        render(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Routes>
                    <Route path="hero/:id" element={ <HeroPage/>}/>
                    <Route path="marvel" element={ <h1>Marvel</h1> }/>
                </Routes>             
            </MemoryRouter>
        )
        
        
        const btnRegresar = screen.getByRole('button');
        fireEvent.click( btnRegresar );

        expect( mockUseNavigate ).toHaveBeenCalledWith(-1);

     })
 })