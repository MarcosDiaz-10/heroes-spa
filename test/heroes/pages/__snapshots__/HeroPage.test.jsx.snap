// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas HeroPage debe mostrar el componente por defecto cuando existe el hero 1`] = `
<div>
  <div
    class="row mt-5"
  >
    <div
      class="col-4"
    >
      <img
        alt="Spider Man"
        class="img-thumbnail animate__animated animate__fadeInLeft"
        src="/assets/marvel-spider.jpg"
      />
    </div>
    <div
      class="col-8 animate__animated animate__zoomIn "
    >
      <h3>
        Spider Man
      </h3>
      <ul
        class="list-group list-group=flush"
      >
        <li
          class="list-group-item"
        >
           
          <b>
            Alter ego:
          </b>
           
          Peter Parker
           
        </li>
        <li
          class="list-group-item"
        >
           
          <b>
            Publisher:
          </b>
           
          Marvel Comics
           
        </li>
        <li
          class="list-group-item"
        >
           
          <b>
            First appearance:
          </b>
           
          Amazing Fantasy #15
           
        </li>
      </ul>
      <h5
        class="mt-3"
      >
         Characters
      </h5>
      <p>
        Peter Parker
      </p>
      <button
        class="btn btn-outline-primary"
      >
        Regresar
      </button>
    </div>
  </div>
</div>
`;
